vertx-web/src/test/java/io/vertx/ext/web/handler/VirtualHostHandlerHttp2Test.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2014 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.web.handler;
+
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.WebTestBase;
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * @author <a href="mailto:emad.albloushi@gmail.com">Emad Alblueshi</a>
+ */
+public class VirtualHostHandlerHttp2Test extends WebTestBase {
+
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    DEFAULT_PORT = 8443;
```

Comment:
Is there a reason why we need to setup SSL here? Since we now both the client and server will user HTTP/2, we can use the `h2c` protocol (non secured connection).